src/v/kafka/server/protocol.cc

```diff
@@ -208,7 +206,7 @@ ss::future<> protocol::apply(net::server::resources rs) {
                     ctx->client_host(),
                     ctx->client_port(),
                     disconnected.value(),
-                    security::sasl_state_to_str(ctx->sasl().state()));
+                    security::sasl_state_to_str(ctx->sasl()->state()));
```

Comment:
given this is (1) in shutdown processing and (2) unconditional, is there any reason to worry about sasl() being cleared during shutdown before we get here?

Reply:
The lifetime of `sasl` is connected to that of `ctx`. I was actually a little concerned under the prospect of reconfiguration, which is why I changed the  condition check above:
```diff
- if (config::shard_local_cfg().enable_sasl()) {
+ if (authn_method == config::broker_authn_method::sasl) {
```